pages/api/v1/child_bounties/getAllChildBounties.ts

```diff
@@ -43,15 +46,59 @@ export const getAllchildBountiesFromBountyIndex = async ({ parentBountyIndex, ne
 			child_bounties_count: subsquidData?.proposalsConnection?.totalCount || 0
 		};
 
-		for (const childBounty of subsquidData.proposals) {
-			resObj.child_bounties.push({
-				description: childBounty.description,
-				index: childBounty.index,
-				reward: childBounty?.reward,
-				status: childBounty.status,
+		const childBountiesProposals = subsquidData?.proposals || [];
+
+		const allChildBountiesIndexes = childBountiesProposals.map((childBounty: { index: number }) => childBounty?.index);
+
+		const chunkArray = (arr: any[], chunkSize: number) => {
+			const chunks = [];
+			for (let i = 0; i < arr.length; i += chunkSize) {
+				chunks.push(arr.slice(i, i + chunkSize));
+			}
+			return chunks;
+		};
+
+		const chunks = chunkArray(allChildBountiesIndexes, 30);
+
+		const childBountiesDocsPromises = chunks.map((chunk) => postsByTypeRef(network, ProposalType.CHILD_BOUNTIES).where('id', 'in', chunk).get());
+
+		const childBountiesDocsSnapshots = await Promise.all(childBountiesDocsPromises);
+
+		const childBountiesDocs = childBountiesDocsSnapshots.flatMap((snapshot) => snapshot.docs);
+
+		console.log(childBountiesDocs);
+
+		const childBountiesPromises = childBountiesProposals?.map(async (subsquidChildBounty: any) => {
+			const payload = {
+				createdAt: subsquidChildBounty?.createdAt,
+				curator: subsquidChildBounty?.curator || '',
+				description: subsquidChildBounty?.description || '',
+				index: subsquidChildBounty?.index,
+				reward: subsquidChildBounty?.reward,
+				status: subsquidChildBounty?.status,
 				title: ''
+			};
+			childBountiesDocs?.map((childBounty) => {
```

Comment:
this will return data for the first child bounty found everytime, not the one we're looping over